autonomy/analyse/service.py

```diff
@@ -36,15 +43,101 @@
 from autonomy.configurations.loader import load_service_config
 
 
-REQUIRED_SETUP_PARAMETERES = (
-    "safe_contract_address",
-    "all_participants",
-)
+ABCI_CONNECTION_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "config": {
+            "type": "object",
+            "required": [
+                "host",
+                "port",
+                "use_tendermint",
+            ],
+        },
+        "required": ["config"],
+    },
+}
 
-REQUIRED_PARAM_VALUES = (
-    "share_tm_config_on_startup",
-    "on_chain_service_id",
-)
+LEDGER_CONNECTION_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "config": {
+            "type": "object",
+            "properties": {
+                "ledger_apis": {
+                    "type": "object",
+                    "properties": {
+                        "ethereum": {
+                            "type": "object",
+                            "properties": {
+                                "address": {},
+                                "chain_id": {},
+                                "poa_chain": {},
+                                "default_gas_price_strategy": {},
+                            },
+                            "required": [
+                                "address",
+                                "chain_id",
+                                "poa_chain",
+                                "default_gas_price_strategy",
+                            ],
+                        },
+                    },
+                    "required": ["ethereum"],
+                }
+            },
+            "required": ["ledger_apis"],
+        },
+    },
+    "required": ["config"],
+}
+
+ABCI_SKILL_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "models": {
+            "type": "object",
+            "properties": {
+                "params": {
+                    "type": "object",
+                    "properties": {
+                        "args": {
+                            "type": "object",
+                            "properties": {
+                                "setup": {
+                                    "type": "object",
+                                    "required": [
+                                        "safe_contract_address",
+                                        "all_participants",
+                                    ],
+                                }
+                            },
+                            "required": [
+                                "setup",
+                                "tendermint_url",
+                                "tendermint_com_url",
+                                "service_registry_address",
+                                "share_tm_config_on_startup",
+                                "on_chain_service_id",
+                            ],
```

Comment:
This is incomplete afaik. Check what we have in scaffolding too. Also the two should be forced to stay in sync

Reply:
We don't want to redefine every single parameter in the service configuration, we need to make sure it contains the basic overrides needed for a deployment not every single override that is defined in the skill config

Reply:
All the overrides defined for the scaffold are needed in each skill. O/w they cannot be over-riden from the agent/service & also the skill instantiation will fail if the concrete skill does not list them. Speak to @Adamantios who can point out all the "enforce" that will raise if various params are not defined.